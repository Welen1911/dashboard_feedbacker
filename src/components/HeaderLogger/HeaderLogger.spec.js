import HeadLogged from "./";
import { routes } from "../../router";

import { createRouter, createWebHistory } from "vue-router";
import { shallowMount } from "@vue/test-utils";

const mockStore = { currentUser: {} };
jest.mock("../../hooks/useStore", () => {
  return () => {
    return mockStore;
  };
});

const router = createRouter({
  history: createWebHistory("/"),
  routes,
});

describe("<HeadLogger />", () => {
  it.skip("should render header logged correctly", () => {
    router.push("/");
    const wrapper = shallowMount(HeadLogged, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it.skip("should render 3 dots when there´s not user logged", () => {
    router.push("/");
    const wrapper = shallowMount(HeadLogged, {
      global: {
        plugins: [router],
      },
    });

    const buttonLogout = wrapper.find("#logout-button");

    expect(buttonLogout.text()).toBe("...");
  });

  it("should render user name when there´s user logged", () => {
    router.push("/");

    mockStore.currentUser.name = "Welen";

    const wrapper = shallowMount(HeadLogged, {
      global: {
        plugins: [router],
      },
    });

    const buttonLogout = wrapper.find("#logout-button");

    expect(buttonLogout.text()).toBe("Welen (Sair)");
  });
});
