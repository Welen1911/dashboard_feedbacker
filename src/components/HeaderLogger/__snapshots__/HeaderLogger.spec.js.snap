// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeadLogger /> should render header logged correctly 1`] = `
<div class="flex justify-center w-full h-28 bg-brand-main">
  <div class="flex items-center justify-between w-4/5 max-w-6xl py-10">
    <div class="w-28 lg:w-36"><img class="w-full" src="" alt="logo"></div>
    <div class="flex">
      <ul class="flex list-none">
        <li class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"> Credenciais </li>
        <li class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"> FeedBacks </li>
        <li id="logout-button" class="px-6 py-2 font-bold bg-white rounded-full cursor-pointer text-brand-main focus:outline-none">...</li>
      </ul>
    </div>
  </div>
</div>
`;
